src/acrcssc/azext_acrcssc/helper/_taskoperations.py

```diff
@@ -145,22 +145,39 @@ def _eval_trigger_run(cmd, registry, resource_group, run_immediately):
         _trigger_task_run(cmd, registry, resource_group, CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME)
 
 
-def delete_continuous_patch_v1(cmd, registry, dryrun):
+def delete_continuous_patch_v1(cmd, registry, yes):
     logger.debug("Entering delete_continuous_patch_v1")
     cssc_tasks_exists, _ = check_continuous_task_exists(cmd, registry)
     cssc_config_exists = check_continuous_task_config_exists(cmd, registry)
-    if not dryrun and (cssc_tasks_exists or cssc_config_exists):
+
+    acr_run_client = cf_acr_runs(cmd.cli_ctx)
+    resource_group_name = parse_resource_id(registry.id)[RESOURCE_GROUP]
+    running_tasks = WorkflowTaskStatus.get_taskruns_with_filter(
+        acr_run_client,
+        registry_name=registry.name,
+        resource_group_name=resource_group_name,
+        status_filter=[TaskRunStatus.Running.value, TaskRunStatus.Queued.value, TaskRunStatus.Started.value],
+        taskname_filter=[CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME, CONTINUOUSPATCH_TASK_SCANIMAGE_NAME, CONTINUOUSPATCH_TASK_PATCHIMAGE_NAME])
+    if len(running_tasks) > 0:
+        from knack.prompting import prompt_y_n
+        if yes or prompt_y_n("There are currently running tasks for this workflow. Do you want to cancel their execution?"):
+            _cancel_task_runs(acr_run_client, registry.name, resource_group_name, running_tasks)
+
+    if cssc_tasks_exists or cssc_config_exists:
         cssc_tasks = ', '.join(CONTINUOUSPATCH_ALL_TASK_NAMES)
         logger.warning(f"All of these tasks will be deleted: {cssc_tasks}")
         for taskname in CONTINUOUSPATCH_ALL_TASK_NAMES:
-            # bug: if one of the deletion fails, the others will not be attempted, we need to attempt to delete all of them
-            _delete_task(cmd, registry, taskname, dryrun)
-            logger.warning(f"Task {taskname} deleted.")
+            try:
+                _delete_task(cmd, registry, taskname)
+                logger.warning(f"Task {taskname} deleted.")
+            except AzCLIError as e:
+                logger.error(f"Failed to delete task {taskname}: {e}")
         logger.warning(f"Deleting {CSSC_WORKFLOW_POLICY_REPOSITORY}/{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG}:{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG_TAG_V1}")
-        delete_oci_artifact_continuous_patch(cmd, registry, dryrun)
+        delete_oci_artifact_continuous_patch(cmd, registry)
```

Comment:
Could you separate the condition check in line 166 and only delete task or config if they exist?